I tried to use this commit from starknet-replay (https://github.com/lambdaclass/starknet-replay/pull/31) but got some compilation errors due to thread safety:

```
error[E0277]: `Rc<AotNativeExecutor>` cannot be sent between threads safely
   --> /Users/julian/.cargo/git/checkouts/blockifier-7ae50b0f3ce149d5/1da9138/crates/blockifier/src/execution/native/utils.rs:79:31
    |
79  | ...OGRAM_CACHE: OnceLock<RwLock<ProgramCache<'static, ClassHash>>> = OnceLock::new();
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Rc<AotNativeExecutor>` cannot be sent between threads safely
```

If you want to replicate it, this is what I did:
- Changed blockifer rev in Cargo.toml to 1da9138b762b2e80ee37edae40a72e434e7c3ec3
- Executed `cargo run --features benchmark bench-block-range 606173 606173 mainnet 1
`

This is the full error message I got:
```
> cargo run --features benchmark bench-block-range 606173 606173 mainnet 1
   Compiling blockifier v0.7.0-dev.1 (https://github.com/lambdaclass/blockifier?rev=1da9138b762b2e80ee37edae40a72e434e7c3ec3#1da9138b)
error[E0277]: `Rc<AotNativeExecutor>` cannot be sent between threads safely
   --> /Users/julian/.cargo/git/checkouts/blockifier-7ae50b0f3ce149d5/1da9138/crates/blockifier/src/execution/native/utils.rs:79:31
    |
79  | ...OGRAM_CACHE: OnceLock<RwLock<ProgramCache<'static, ClassHash>>> = OnceLock::new();
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Rc<AotNativeExecutor>` cannot be sent between threads safely
    |
    = help: within `(starknet_api::core::ClassHash, Rc<AotNativeExecutor>)`, the trait `Send` is not implemented for `Rc<AotNativeExecutor>`, which is required by `OnceLock<std::sync::RwLock<ProgramCache<'static, starknet_api::core::ClassHash>>>: Sync`
    = note: required because it appears within the type `(starknet_api::core::ClassHash, Rc<AotNativeExecutor>)`
    = note: required for `hashbrown::raw::RawTable<(starknet_api::core::ClassHash, Rc<AotNativeExecutor>)>` to implement `Send`
note: required because it appears within the type `hashbrown::map::HashMap<starknet_api::core::ClassHash, Rc<AotNativeExecutor>, RandomState>`
   --> /rust/deps/hashbrown-0.14.3/src/map.rs:190:12
note: required because it appears within the type `HashMap<starknet_api::core::ClassHash, Rc<AotNativeExecutor>>`
   --> /Users/julian/.rustup/toolchains/1.78.0-aarch64-apple-darwin/lib/rustlib/src/rust/library/std/src/collections/hash/map.rs:213:12
    |
213 | pub struct HashMap<K, V, S = RandomState> {
    |            ^^^^^^^
note: required because it appears within the type `AotProgramCache<'static, starknet_api::core::ClassHash>`
   --> /Users/julian/.cargo/git/checkouts/cairo_native-0bad8506281b7281/66e9b5e/src/cache/aot.rs:14:12
    |
14  | pub struct AotProgramCache<'a, K>
    |            ^^^^^^^^^^^^^^^
note: required because it appears within the type `ProgramCache<'static, starknet_api::core::ClassHash>`
   --> /Users/julian/.cargo/git/checkouts/cairo_native-0bad8506281b7281/66e9b5e/src/cache.rs:8:10
    |
8   | pub enum ProgramCache<'a, K>
    |          ^^^^^^^^^^^^
    = note: required for `std::sync::RwLock<ProgramCache<'static, starknet_api::core::ClassHash>>` to implement `Sync`
    = note: 1 redundant requirement hidden
    = note: required for `OnceLock<std::sync::RwLock<ProgramCache<'static, starknet_api::core::ClassHash>>>` to implement `Sync`
    = note: shared static variables must have a type that implements `Sync`
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,7 +27,7 @@ cairo-lang-sierra = "2.6.4"
 cairo-lang-starknet = "2.6.4"
 cairo-lang-starknet-classes = "2.6.4"
 cairo-lang-utils = "2.6.4"
-cairo-native = { git = "https://github.com/lambdaclass/cairo_native", rev = "905d6959662edfec4829568f995206fcb8bf64d0" } # main
+cairo-native = { git = "https://github.com/lambdaclass/cairo_native", rev = "72d10b7923cb4f97ae0b1688ea7de1d34297b0b2" } # main
 cairo-vm = "0.9.2"
 criterion = "0.3"
 derive_more = "0.99.17"

